plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProvider.java

```diff
@@ -389,23 +459,18 @@ private VFSFile doMove( VFSFile file, String newPath, OverwriteStatus overwriteS
           return null;
         } else if ( overwriteStatus.isRename() ) {
           VFSDirectory vfsDir =
-            VFSDirectory.create( renameObject.getParent().getPath().toString(), renameObject, file.getConnection(),
-              file.getDomain() );
+            createVFSDirectory( renameObject.getParent().getPath().toString(), renameObject );
           newPath = getNewName( vfsDir, newPath, space  );
-          renameObject = KettleVFS
-            .getFileObject( newPath, new Variables(), VFSHelper.getOpts( bowl, file.getPath(), file.getConnection(),
-                                                                         space ) );
+          renameObject = getFileObject( newPath, space );
         }
       }
       fileObject.moveTo( renameObject );
       if ( file instanceof VFSDirectory ) {
-        return VFSDirectory.create( renameObject.getParent().getPublicURIString(), renameObject, file.getConnection(),
-          file.getDomain() );
+        return createVFSDirectory( renameObject.getParent().getPublicURIString(), renameObject );
```

Comment:
No change needed.
I know this was already there, but I wonder why some instances do `renameObject.getParent().getPath().toString()` instead.
Smells...

Reply:
yeah, I saw this. I can't solve everything. just trying to focus on the conversion to PVFS URI. There's many code smells in this part of the code.